engine/src/cmd/planet.cpp

```diff
@@ -317,30 +314,28 @@ Planet::Planet(QVector x,
         }
     }
     if (ligh.size() > 0) {
-        static float
-                bodyradius = XMLSupport::parse_float(vs_config->getVariable("graphics", "star_body_radius", ".33"));
-        static bool drawglow = XMLSupport::parse_bool(vs_config->getVariable("graphics", "draw_star_glow", "true"));
-
-        static bool drawstar = XMLSupport::parse_bool(vs_config->getVariable("graphics", "draw_star_body", "true"));
-        static float glowradius =
-                XMLSupport::parse_float(vs_config->getVariable("graphics", "star_glow_radius", "1.33")) / bodyradius;
+        const float bodyradius = configuration()->graphics_config.star_body_radius;
+        const bool drawglow = configuration()->graphics_config.draw_star_glow;
+        const bool drawstar = configuration()->graphics_config.draw_star_body;
+        const float glowradius = configuration()->graphics_config.star_glow_radius / bodyradius;
         if (drawglow) {
             GFXColor c = getMaterialEmissive(ourmat);
-            static bool spec =
-                    XMLSupport::parse_bool(vs_config->getVariable("graphics", "glow_ambient_star_light", "false"));
-            static bool diff =
-                    XMLSupport::parse_bool(vs_config->getVariable("graphics", "glow_diffuse_star_light", "false"));
+            const bool spec = configuration()->graphics_config.glow_ambient_star_light;
+            const bool diff = configuration()->graphics_config.glow_diffuse_star_light;
             if (diff) {
                 c = ligh[0].ligh.GetProperties(DIFFUSE);
             }
             if (spec) {
                 c = ligh[0].ligh.GetProperties(AMBIENT);
             }
-            static vector<string>
+
+            // TODO: Refactor the below section of code
+            vector<string>
                     shines = ParseDestinations(vs_config->getVariable("graphics", "star_shine", "shine.ani"));
             if (shines.empty()) {
-                shines.push_back("shine.ani");
+                shines.emplace_back("shine.ani");
             }
```

Comment:
would this ever be true? shouldn't the new config stuff make sure that `shine.ani` is provided as a default when read from the configuration?

Reply:
Yeah, this would probably only be true if there was a setting present in the config file that specifically set this to an empty/null value. I figure it's better to be safe than sorry.